server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4927,11 +4927,10 @@ public void finalizeStop(VirtualMachineProfile profile, Answer answer) {
         if (owner.getState() == Account.State.disabled) {
             throw new PermissionDeniedException("The owner of " + vm + " is disabled: " + vm.getAccountId());
         }
-        if (VirtualMachineManager.ResoureCountRunningVMsonly.value()) {
-            // check if account/domain is with in resource limits to start a new vm
-            ServiceOfferingVO offering = _serviceOfferingDao.findById(vm.getId(), vm.getServiceOfferingId());
-            resourceLimitCheck(owner, vm.isDisplayVm(), new Long(offering.getCpu()), new Long(offering.getRamSize()));
-        }
+
```

Comment:
@Pearl1594 
I think we need to check resource limitation and update resource count when update the vm setting.

Reply:
do we need to keep this ?

Reply:
Based on my understanding, operations like scaleVM - which could update the vm settings prior starting the VM, and other such operations, have these checks in place, so it may become redundant. However, keeping this check may be safer